bindings/java/src/java_module_host.c

```diff
@@ -239,8 +248,41 @@ static MODULE_HANDLE JavaModuleHost_Create(BROKER_HANDLE broker, const void* con
                                                 }
                                                 else
                                                 {
-                                                    jobject jModule_object = NewObjectInternal(result->env, jModule_class, jModule_constructor, 3, (jlong)result, jBroker_object, jModule_configuration);
-                                                    exception = JNIFunc(result->env, ExceptionOccurred);
+                                                    jobject jModule_object;
+                                                    if (emptyConstructor)
+                                                    {
+                                                        jModule_object = NewObjectInternal(result->env, jModule_class, jModule_constructor, 0);
```

Comment:
The unit tests might be tricky to sift through if you're looking at them completely fresh, let me know if I can help with that.